apps/native/app/src/screens/document-detail/utils/share-file.tsx

```diff
@@ -0,0 +1,34 @@
+import Share from 'react-native-share'
+
+import { isAndroid } from '../../../utils/devices'
+import { authStore } from '../../../stores/auth-store'
+import { DocumentV2 } from '../../../graphql/types/schema'
+
+interface ShareFileProps {
+  document: DocumentV2
+  hasPdf: boolean
+  pdfUrl?: string
+}
+
+export const shareFile = ({ document, hasPdf, pdfUrl }: ShareFileProps) => {
+  if (
+    !document ||
+    !document.subject ||
+    !document.sender ||
+    (hasPdf && !pdfUrl)
+  ) {
+    return
```

Comment:
Is the app logging unexplained events? Maybe this place would be good to log that something is missing 🤔